### PR TITLE
chore: switch from `yarn` to `pnpm`

### PR DESCRIPTION
**Pull request by coadler**

### CODE REVIEW

**Comment by matifali** [comment]
Will it reduce the setup-node time?

**Comment by coadler** [comment]
> Will it reduce the setup-node time?

Ideally, yes.

**Review by BrunoQuaresma — APPROVED** [review]
I like this 👍

File: offlinedocs/.gitignore [hunk 1]
```diff
@@ -23,7 +23,7 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
```
Comment (matifali) [inline offlinedocs/.gitignore:25]:
```suggestion

```
also here

File: flake.nix [hunk 2]
```diff
@@ -49,6 +49,7 @@
             terraform
             typos
             yarn
```
Comment (matifali) [inline flake.nix:51]:
```suggestion

```
Should we remove yarn from here?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+devDependencies:
+  prettier:
+    specifier: 3.0.0
+    version: 3.0.0
+
+packages:
+
+  /prettier@3.0.0:
+    resolution: {integrity: sha512-zBf5eHpwHOGPC47h0zrPyNn+eAEIdEzfywMoYn2XPi0P44Zp0tSq64rq0xAREh4auw2cJZHo9QUob+NqCQky4g==}
+    engines: {node: '>=14'}
```
Comment (matifali) [inline pnpm-lock.yaml:16]:
```suggestion
    engines: {node: '>=18'}
```
Shouldn't it be greater than >=18 as now we have moved to node 18.

Comment (BrunoQuaresma) [reply pnpm-lock.yaml:16]: And less than 19.0.0

Comment (coadler) [reply pnpm-lock.yaml:16]: This is just showing the node compatibility of the package.

Comment (BrunoQuaresma) [reply pnpm-lock.yaml:16]: lol that is true 🤦‍♂️

**Pull request merged**